### ISSUE TITLE
Consider switching to spft13/afero instead of using go-billy

### ISSUE THREAD

**Issue by thefallentree** [src-d/go-git#315]
Hi,

Looks like https://github.com/spf13/afero is a more mature library that basically does the samething with go-billy.  Please consider switching to that instead, should be pretty easy.

**Comment by mcuadros** [comment]
Duplicate https://github.com/src-d/go-git/issues/155

**Comment by mcuadros** [comment]
But maybe we can bring the topic again internally, what do you think @smola?

**Comment by smola** [comment]
@thefallentree Thanks for bringing this up. Last time we discussed this, we decided to keep using go-billy, since its API is more convenient for some of our use cases (working directly on file paths, without any consideration about directories). However, we'll re-evaluate this, since we might achieve the same with a thin wrapper FS for afero.

**Comment by smola** [comment]
We evaluated this again. This is what we considered:

* **Maintenance:** afero does not seem to have too active development.
* **Demand:** we got the suggestion of using afero from multiple people, but so far nobody had the need to use an afero implementation with go-git, nor the intention to write a new one.
* **API:** afero API is indeed more convenient in some sense: it mimicks the standard library, so it might be more intuitive for new users. It is actually more feature-complete than go-billy's.
* **Implementations:** there doesn't seem to be many afero implementations in the wild, but we do have a few go-billy implementations (to be released as open source soon).

We decided not to switch. We'll continue development with go-billy and will release more implementations of it. Further major versions of go-billy might actually use an interface more similar to the standard library (so, more similar to afero too).

Closing this issue.
